src/widgets/helper/ChannelView.cpp

```diff
@@ -2557,20 +2557,36 @@ void ChannelView::addMessageContextMenuItems(QMenu *menu,
     if (this->canReplyToMessages() && layout->isReplyable())
     {
         const auto &messagePtr = layout->getMessagePtr();
-        menu->addAction("&Reply to message", [this, &messagePtr] {
-            this->setInputReply(messagePtr);
-        });
-
         if (messagePtr->replyThread != nullptr)
         {
-            menu->addAction("Reply to &original thread", [this, &messagePtr] {
-                this->setInputReply(messagePtr->replyThread->root());
-            });
+            const auto &rootPtr = messagePtr->replyThread->root();
+            if (!rootPtr->flags.has(MessageFlag::InvalidReplyTarget))
+            {
+                // if the root of the thread is an invalid reply target (i.e. has been deleted)
+                // the whole thread is now invalid reply target (Twitch behaviour)
```

Comment:
This should be above the `if` - it looks like this is for the `messagePtr->flags.has(MessageFlag::InvalidReplyTarget)` check.

Reply:
moved